lib/fizzy/leb128.hpp

```diff
@@ -46,13 +52,30 @@ std::pair<T, const uint8_t*> leb128s_decode(const uint8_t* input, const uint8_t*
         result |= static_cast<T_unsigned>((static_cast<T_unsigned>(*input) & 0x7F) << result_shift);
         if ((*input & 0x80) == 0)
         {
-            // sign extend
-            if ((*input & 0x40) != 0 && (result_shift + 7 < sizeof(T_unsigned) * 8))
+            constexpr auto all_ones = std::numeric_limits<T_unsigned>::max();
+
+            if (result_shift + 7 < sizeof(T_unsigned) * 8)
             {
-                constexpr auto all_ones = std::numeric_limits<T_unsigned>::max();
```

Comment:
This is still only used in this branch, however it should make no difference.

Reply:
Moved it back to the branch where it's only needed.